src/popover/body.tsx

```diff
@@ -34,12 +31,12 @@ export default function PopoverBody({
   children,
   onDismiss,
   variant,
-  returnFocus = true,
   overflowVisible,
-  dismissButtonRef,
   className,
 }: PopoverBodyProps) {
   const labelledById = useUniqueId('awsui-popover-');
+  const [dismissButtonFocused, setDismissButtonFocused] = useState(false);
```

Comment:
As this state is only used inside an effect it can be implemented with a ref to avoid unnecessary re-render:

```
const dismissButtonFocusedRef = useRef(false)

useEffect(() => {
  if (showDismissButton && !dismissButtonFocusedRef.current) {
    dismissButtonRef.current?.focus({ preventScroll: true });
  }
  dismissButtonFocusedRef.current = showDismissButton
}, [showDismissButton])
```